File: common/src/main/java/cc/turtl/chiselmon/api/filter/match/FilterMatchResult.java
package cc.turtl.chiselmon.api.filter.match;

import cc.turtl.chiselmon.api.filter.RuntimeFilter;
import com.cobblemon.mod.common.pokemon.Pokemon;

import java.util.List;
import java.util.Optional;

public record FilterMatchResult(
        Pokemon pokemon,
        Optional<RuntimeFilter> primaryMatch,
        List<RuntimeFilter> allMatches
) {
}

File: common/src/main/kotlin/cc/turtl/chiselmon/core/api/filter/match/FilterMatchResult.kt
package cc.turtl.chiselmon.core.api.filter.match

import cc.turtl.chiselmon.core.api.filter.RuntimeFilter
import com.cobblemon.mod.common.pokemon.Pokemon

data class FilterMatchResult(
    @JvmField val pokemon: Pokemon,
    @JvmField val primaryMatch: RuntimeFilter?,
    @JvmField val allMatches: List<RuntimeFilter>
)